docs/user-guide/examples.md

```diff
@@ -130,4 +130,35 @@ defaultEntryPoints = ["http"]
     headerField = "X-WebAuth-User"
     [entryPoints.http.auth.basic]
     users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/", "test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"]
+```
+
+
+## Enable authentication forwarding
+
+Uses a query param from the request to check for authentication with an auth service.
+If the auth service returns 200 OK the request is forwarded to the backend with additional headers/query params from the auth service.
+The response from the auth service is parsed, json encoded and can be forwarded to the backend as headers/query params.
+The field names in the request and the response can be manipulated for internal use via `as` config for request params 
+and the response from the auth server can be manipulated into a different name via `as` config for the response params.
+```
+defaultEntryPoints = ["http"]
+[entryPoints]
+  [entryPoints.http]
+  address = ":80"
+  [entryPoints.http.auth.forward]
+    address = "http://authserver.com/auth"
+    [entryPoints.http.auth.Forward.requestParameters.email]
+      name = "email"
+      as = "theEmail"
+    [entryPoints.http.auth.Forward.requestParameters.token]
+      name = "token"
+      as = "theToken"
+    [entryPoints.http.auth.Forward.responseReplayFields.userId]
+      path = "user.id"
+      as = "X-User-Id"
+      in = "header"
+    [entryPoints.http.auth.Forward.responseReplayFields.userName]
+      path = "user.name"
+      as = "" # No name transformation
+      in = "parameter"
```

Comment:
I didn't quite get it so I (tried) to read the source code an rewrote it a bit, what do you think of this (please also make sure it's correct what I'm saying :D):

Proposal:

> Authentication forwarding is a mechanism to allow for arbitrary authentication back ends to be implemented. It works by forwarding any incoming request from traefik proxy to a specified endpoint (the setting key `entryPoints.http.auth.forward.address`). This endpoint address is called "authentication back end". The response of the authentication back end is then evaluated and depending on the HTTP status code the original request to traefik is either allowed or denied. The authentication back end has to send a response with an HTTP status code `200` to allow access. Any other status code will be used as response status code by traefik to the original request (so if your back end sends a response with `503`, traefik will also send a `503` response).

> In addition to this, traefik provides basic functionality to modify the original request's query parameters before they are sent to the authentication back end. This allows you to map a request parameter (in this case `token`) from e.g. `traefik.com/secret?token=foobar` to `authserver.com/auth?theToken=foobar`. Use the keywords `name` and  `as` in the `entryPoints.http.auth.Forward.requestParameters` setting key as shown in the example.

> Moreover, you can also replay certain information from the back end authentication response back to the original request received by traefik. For this to work, your authentication back end must send a JSON response.

Now I'm lost a bit because I would like to add the information about replaying back end auth information and I understand the `header` option but can you explain the `parameter` option for me?